src/BugsnagUnity/Client.cs

```diff
@@ -639,5 +643,91 @@ public void ClearFeatureFlags()
             _featureFlags.Clear();
             NativeClient.ClearFeatureFlags();
         }
+
+        private void SetupNetworkListeners()
+        {
+            BugsnagUnityWebRequest.OnSend.AddListener(OnWebRequestSend);
+            BugsnagUnityWebRequest.OnComplete.AddListener(OnWebRequestComplete);
+            BugsnagUnityWebRequest.OnAbort.AddListener(OnWebRequestAbort);
+        }
+
+        private readonly Dictionary<BugsnagUnityWebRequest, DateTimeOffset> _requestStartTimes = new Dictionary<BugsnagUnityWebRequest, DateTimeOffset>();
+
+
+        private void OnWebRequestComplete(BugsnagUnityWebRequest request)
+        {
+            if (_requestStartTimes.TryGetValue(request, out DateTimeOffset startTime))
+            {
+                TimeSpan duration = DateTimeOffset.UtcNow - startTime;
+                LeaveNetworkBreadcrumb(request.UnityWebRequest, duration);
+                _requestStartTimes.Remove(request);
+            }
+            else
+            {
+                LeaveNetworkBreadcrumb(request.UnityWebRequest, null);
+            }
+        }
+
+        private void OnWebRequestSend(BugsnagUnityWebRequest request)
+        {
+            _requestStartTimes[request] = DateTimeOffset.UtcNow;
+        }
+
+        private void OnWebRequestAbort(BugsnagUnityWebRequest request)
+        {
+            if (_requestStartTimes.ContainsKey(request))
+            {
+                _requestStartTimes.Remove(request);
+            }
+        }
+
+        public void LeaveNetworkBreadcrumb(UnityWebRequest request, TimeSpan? duration)
+        {
+            if(!Configuration.IsBreadcrumbTypeEnabled(BreadcrumbType.Request))
```

Comment:
In the spec we say that the config option controls whether automatic breadcrumbs are set and should control whether we listen to events, rather than whether the breadcrumb is left:

> enabledBreadcrumbTypes shall define which system events the notifier will register/listen for in order to collect Breadcrumbs automatically.

That means that if the developer disables the type, we're not listening at all to the events (i.e. there's no overhead) rather than listen but do nothing.

(That probably needs to be applied to navigation breadcrumbs above too.)

Reply:
i understand the idea, but it's our wrapper so if they use the wrapper then we listen to the events.
also, we use the navigation events for more than breadcrumbs, so it doesn't really make sense in these cases i think?

Reply:
Done for the network listeners, not possible to check the navigation config any earlier